pool/abortable.go

```diff
@@ -123,21 +124,29 @@ func (p *abortablePool) worker(ctx context.Context, jobs <-chan amboy.Job) {
 				job.AddError(err)
 				p.queue.Complete(ctx, job)
 			}
+
 			// start a replacement worker.
 			go p.worker(ctx, jobs)
 		}
+
+		if cancel != nil {
```

Comment:
How can cancel not be nil here?

Reply:
if the jobs channel only gets nil jobs, and then the context is canceled, this will run before cancel is set and we panic.